mlflow/utils/validation.py

```diff
@@ -386,3 +390,22 @@ def _validate_db_type_string(db_type):
     if db_type not in DATABASE_ENGINES:
         error_msg = "Invalid database engine: '%s'. '%s'" % (db_type, _UNSUPPORTED_DB_TYPE_MSG)
         raise MlflowException(error_msg, INVALID_PARAMETER_VALUE)
+
+
+def _validate_model_version_or_stage_exists(version, stage):
+    if version and stage:
+        raise MlflowException("version and stage cannot be set together", INVALID_PARAMETER_VALUE)
+
+    if not (version or stage):
+        raise MlflowException("version or stage must be set", INVALID_PARAMETER_VALUE)
+
+    if stage and stage not in ALL_STAGES:
+        raise MlflowException(
+            "Invalid stage value: %s. Must be one of %s" % (stage, ", ".join(ALL_STAGES)),
+            INVALID_PARAMETER_VALUE,
+        )
```

Comment:
I realize that we do this kind of client-side stage validation in other parts of the MLflow client, but I think it's best to let the backend validate the stage. Can we remove this block?

Reply:
I understand what you mean. The error I see when I use invalid stage value is `MlflowException: Invalid Model Version stage a.` Can I improve the error msg to be `Invalid stage value: a. Value must be one of None, Staging, Production, Archived`